app/components/theme/switcher_component.html.erb

```diff
@@ -1,4 +1,46 @@
-<%= link_to themes_path(theme: other_theme.name), class: yass(link: type), data: { turbo_method: :put } do %>
-  <%= inline_svg_tag "icons/#{current_theme.icon}.svg", class: yass(icon: type), aria: true, title: 'theme icon' %>
-  <%= text unless icon_only? %>
-<% end %>
+<div class="relative <% mobile? ? 'dark:hover:bg-gray-700/60 dark:hover:text-gray-200' : '' %>" data-controller="visibility" data-action="visibility:click:outside->visibility#off" data-visibility-visible-value="false">
+  <% if mobile? %>
```

Comment:
A dropdown within the slide out menu feels a little off to me, one too many layers imo. But I think a select box might work better with smaller screens. We could reuse the same style of select input we've added in the [sortable work](https://github.com/TheOdinProject/theodinproject/blob/main/app/components/sort_component.html.erb) here.